### ISSUE TITLE
Build problems (with GTK 4.10?)

### ISSUE THREAD

**Issue by jackhill** [dino/dino#1479]
Hi Dino,

I recently tried to build Dino from a [Guix branch](https://git.savannah.gnu.org/cgit/guix.git/log/?h=gnome-team)  with some GTK and GNOME updates. In particular, GTK has been updated from 4.8.1 to 4.10.3. That results in the following errors during the build:

```
/tmp/guix-build-dino-0.4.3.drv-0/dino-0.4.3/main/src/ui/util/helper.vala:180.26-180.33: error: Argument 1: Cannot convert from `unowned uint8[]' to `unowned string'
  180 |         p.load_from_data(css.data);
      |                          ^~~~~~~~  
/tmp/guix-build-dino-0.4.3.drv-0/dino-0.4.3/main/src/ui/util/helper.vala:180.9-180.34: error: 1 missing arguments for `void Gtk.CssProvider.load_from_data (string, ssize_t)'
  180 |         p.load_from_data(css.data);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~ 
```

Let me know what additional information might be helpful. I've attached a [gzipped full build log](https://github.com/dino/dino/files/12589427/3p66d4y1mfixplqqh9s5dd5hsrnpkb-dino-0.4.3.drv.gz).

Best,
Jack

**Comment by mar-v-in** [comment]
This is a bug in Vala 0.56.11/0.56.12 and was fixed in 0.56.13.

Upstream issue: https://gitlab.gnome.org/GNOME/vala/-/issues/1478

**Comment by jackhill** [comment]
Indeed, updating Vala solved it. Thanks for the information, and sorry for the noise. I've submitted a [patch to have Vala updated in Guix](https://issues.guix.gnu.org/65901). I'll close this issue. Thanks again!
